tests/unit/index-nodetest.js

```diff
@@ -42,11 +50,41 @@ describe('the index', function() {
               type: 'file-hash',
               filePattern: 'eeee'
             }
+          },
+          commandLineArgs: {}
+        }
+      };
+
+      return assert.isFulfilled(plugin.configure.call(plugin, context));
+    });
+
+    it('sets the revision if it exists in the command line args', function() {
+      var plugin = subject.createDeployPlugin({
+        name: 'revision-key'
+      });
+
+      var context = {
+        deployment: {
+          ui: {
+            write: function() {},
+            writeLine: function() {}
+          },
+          config: {
+            'revision-key': {
+              type: 'file-hash',
+              filePattern: 'eeee'
+            }
+          },
+          commandLineArgs: {
+            revision: 'some-revision'
           }
         }
       };
 
       return assert.isFulfilled(plugin.configure.call(plugin, context))
+        .then(function(result) {
+          assert.equal(result.revision, 'some-revision');
```

Comment:
It doesn't fulfil with the config object....If a hook returns an object it will be merged back in to the context....So, if the commandLineArgs contains the revision, the `configure` hook will be merging it into the context which is where the `activate` hook will reference it from.  Isn't that how we were discussing it would work?